File: system-be/SystemGame.Api/Data/Entities/Agent.cs
namespace SystemGame.Api.Data.Entities;

public class Agent
{
    public int Id { get; set; }
    public string PlayerId { get; set; } = string.Empty;
    public int GameId { get; set; }
    public AgentType Type { get; set; }
    public string Name { get; set; } = string.Empty;
    public AgentState State { get; set; } = AgentState.Idle;
    public string? CurrentBehaviorName { get; set; }
    public string? BehaviorConfig { get; set; } // JSON configuration
    public DateTime LastExecutionTime { get; set; } = DateTime.UtcNow;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Location tracking (nullable - not all agents have locations)
    public int? CurrentSystemId { get; set; }
    public int? CurrentPlanetId { get; set; }
    public double? PositionX { get; set; }
    public double? PositionY { get; set; }
    
    // Entity references (polymorphic)
    public int? BuilderId { get; set; }
    public int? SpaceshipId { get; set; } // Future: Phase 7
    
    // Navigation properties
    public virtual AppUser Player { get; set; } = null!;
    public virtual Game Game { get; set; } = null!;
    public virtual StarSystem? CurrentSystem { get; set; }
    public virtual Planet? CurrentPlanet { get; set; }
    public virtual Builder? Builder { get; set; }
    public virtual ICollection<AgentLog> Logs { get; set; } = new List<AgentLog>();
}

File: system-fe/lib/types/agent.ts
export interface Agent {
  id: number;
  playerId: string;
  gameId: number;
  type: string;
  name: string;
  state: string;
  currentBehaviorName?: string;
  behaviorConfig?: string;
  lastExecutionTime: string;
  createdAt: string;
  currentSystemId?: number;
  currentSystemName?: string;
  currentPlanetId?: number;
  currentPlanetName?: string;
  builderId?: number;
  builderName?: string;
}

export interface CreateAgentRequest {
  gameId: number;
  type: string;
  name: string;
  behaviorName?: string;
  behaviorConfig?: string;
  builderId?: number;
}

export interface UpdateAgentRequest {
  name?: string;
  behaviorName?: string;
  behaviorConfig?: string;
}

export interface AgentLog {
  id: number;
  timestamp: string;
  level: string;
  message: string;
  data?: string;
}

export interface BehaviorInfo {
  name: string;
  description: string;
  supportedAgentTypes: string[];
  configSchema?: string;
}

export const AgentTypes = {
  BUILDER: 'Builder',
  RESOURCE_FERRY: 'ResourceFerry',
  SCOUT: 'Scout',
  FLEET: 'Fleet',
  CUSTOM: 'Custom',
} as const;

export const AgentStates = {
  IDLE: 'Idle',
  ACTIVE: 'Active',
  PAUSED: 'Paused',
  ERROR: 'Error',
  COMPLETED: 'Completed',
} as const;
